src/plugins/nova_correlation_plugin.erl

```diff
@@ -7,11 +7,13 @@
 
 -include_lib("kernel/include/logger.hrl").
 
-pre_request(Req, _) ->
-    UUID = uuid:uuid_to_string(uuid:get_v4()),
+pre_request(Req, Opts) ->
+    CorrId = get_correlation_id(Req, Opts),
     %% Update the loggers metadata with correlation-id
-    logger:update_process_metadata(#{<<"correlation-id">> => UUID}),
-    Req1 = cowboy_req:set_resp_header(<<"X-Correlation-ID">>, UUID, Req),
+    ok = update_logger_metadata(CorrId, Opts),
+    %% Redundant entry, allow controllers to erlang:get(correlation_id)
+    ok = put_process_dictionary(CorrId, Opts),
```

Comment:
It's probably a better idea to set a key in the Req object instead for keeping this in the process-dictionary.

Reply:
changed so that it is set as the `correlation_id` field in the request object.